src/ast/desugar.rs

```diff
@@ -11,8 +10,6 @@ pub struct Desugar {
 impl Default for Desugar {
     fn default() -> Self {
         Self {
-            // the default reserved string in egglog is "_"
-            fresh_gen: SymbolGen::new("_".repeat(2)),
```

Comment:
The current codebase uses a different default symbol for desugar ("__") and typechecking ("$"). Is it okay to merge them into a single fresh generator? I think while this has no soundness issue, we (either users or developers of egglog) no longer know the provenance of internally generated variables.

Reply:
We should use the same distinguishing symbol ("$") for both
If we want to know the provenance, we can change the name hint